lib/schema/utils.js

```diff
@@ -389,9 +403,11 @@ const combineErrors = ({ errorCache, warningsCache, payloadErrors, stats, versio
               count: details.occurrences,
               ...(file && { fileName: file }),
               ...(pathName && { pathName }),
-              ...(lookupValue && { lookupValue }),
+              ...(details?.failedEnum != null && { lookupValue }),
               ...(details?.sourceModel && { sourceModel: details.sourceModel }),
               ...(details?.sourceModelField && { sourceModelField: details.sourceModelField }),
+              ...(details?.keyField && { keyField: details.keyField }),
```

Comment:
Probably better if you don't use fall-through evaluation of `&&` to return the value here.